plugins/com.google.cloud.tools.eclipse.projectselector/src/com/google/cloud/tools/eclipse/projectselector/DefaultProjectRepository.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.projectselector;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson.JacksonFactory;
+import com.google.api.services.cloudresourcemanager.CloudResourceManager;
+import com.google.api.services.cloudresourcemanager.model.ListProjectsResponse;
+import com.google.api.services.cloudresourcemanager.model.Project;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Default implementation of {@link ProjectRepository}.
+ */
+public class DefaultProjectRepository implements ProjectRepository {
+
+  private static final int PROJECT_LIST_PAGESIZE = 300;
+  private static final String PROJECT_DELETE_REQUESTED = "DELETE_REQUESTED";
+
+  @Override
+  public List<GcpProject> getProjects(Credential credential) throws ProjectRepositoryException {
+    // TODO cache results https://github.com/GoogleCloudPlatform/google-cloud-eclipse/issues/1374 
+    try {
+      if (credential != null) {
+        JsonFactory jsonFactory = new JacksonFactory();
+        HttpTransport transport = new NetHttpTransport();
+        CloudResourceManager resourceManager =
+            new CloudResourceManager.Builder(transport, jsonFactory, credential)
+              .setApplicationName("google-cloud=eclipse").build();
```

Comment:
We need to be consistent on the name. I think we can use `CloudToolsInfo.METRICS_NAME` or `CloudToolsInfo.USER_AGENT`, which we are already using for sending Analytics pings.